src/lib.rs

```diff
@@ -1384,6 +1369,165 @@ fn digraph_all_simple_paths(
     Ok(result)
 }
 
+fn weight_callable(
+    py: Python,
+    weight_fn: &Option<PyObject>,
+    weight: &PyObject,
```

Comment:
My intuition tells me this `&` addition makes sense, and it must be related to using `e.weight()` below, but I'm not sure why. Can you explain to me why you added it?

Reply:
Does `e.weight()` not return a value?

Reply:
`e.weight()` returns a `&PyObject` a reference/borrow of a PyObject. So to avoid needing to increment the python ref counter and creating a separate object I made this change so we can just pass the reference to python directly.